pkg/nsx/services/ipblocksinfo/ipblocksinfo.go

```diff
@@ -371,32 +382,65 @@ func (s *IPBlocksInfoService) getIPBlockCIDRsByVPCConfig(vpcConfigList []v1alpha
 	queryParam = fmt.Sprintf("%s:%s", common.ResourceType, common.ResourceTypeIPBlock)
 	count, err = s.SearchResource(common.ResourceTypeIPBlock, queryParam, ipBlockStore, nil)
 	if err != nil {
-		return externalIPCIDRs, privateTGWIPCIDRs, err
+		return
 	}
 	log.V(2).Info("successfully fetch all IPBlocks from NSX", "count", count)
 
-	if externalIPCIDRs, err = s.getIPBlockCIDRsFromStore(externalIPBlockPaths, ipBlockStore); err != nil {
-		return nil, nil, err
-	}
-	if privateTGWIPCIDRs, err = s.getIPBlockCIDRsFromStore(privateTgwIPBlockPaths, ipBlockStore); err != nil {
-		return nil, nil, err
-	}
-	return externalIPCIDRs, privateTGWIPCIDRs, nil
+	externalIPCIDRs, privateTGWIPCIDRs, externalIPRanges, privateTGWIPRanges = s.getCIDRsRangesFromStore(externalIPBlockPaths, privateTgwIPBlockPaths, ipBlockStore)
+	return externalIPCIDRs, privateTGWIPCIDRs, externalIPRanges, privateTGWIPRanges, nil
 }
 
-func (s *IPBlocksInfoService) getIPBlockCIDRsFromStore(pathSet sets.Set[string], ipBlockStore *IPBlockStore) ([]string, error) {
+func (s *IPBlocksInfoService) getCIDRsRangesFromStore(externalIPBlockPaths, privateTgwIPBlockPaths sets.Set[string], ipBlockStore *IPBlockStore) (externalIPCIDRs, privateTGWIPCIDRs []string, externalIPRanges, privateTGWIPRanges []v1alpha1.IPPoolRange) {
```

Comment:
func getCIDRsRangesFromStore is not necessary?

Reply:
removed

Reply:
Remember to remove this function in a following patch